File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # 가로/세로 구분 x, 짧은 부분/긴 부분
    # -> 각 부분의 max를 곱함
    
    width = []
    height = []
    
    for size in sizes:
        width.append(max(size[0], size[1]))
        height.append(min(size[0], size[1]))
    
    return max(width) * max(height)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int smaller = Integer.MIN_VALUE;
        int bigger = Integer.MIN_VALUE;
        
        for (int[] size: sizes) {
            if (size[0] < size[1]) {
                smaller = Math.max(smaller, size[0]);
                bigger = Math.max(bigger, size[1]);
            } else {
                smaller = Math.max(smaller, size[1]);
                bigger = Math.max(bigger, size[0]);
            }
        }
        
        return smaller * bigger;
    }
}